src/nnet3/nnet-nnet.h

```diff
@@ -229,6 +230,13 @@ class Nnet {
 
   // Assignment operator
   Nnet& operator =(const Nnet &nnet);
+
+  void ResetGenerators(); // resets random-number generators for all
+  // random components.  You must also set sRand() for this to be
+  // effective.
+  
+  // Removes all dropout components from network, which can be used during test time.
+  void RemoveDropout();
```

Comment:
Can you please take out this `RemoveDropout()` function?
I have been thinking about it, and I think the best way to do this would be at the config level: a file "remove-ephemeral-connections.config" would be created that modifies the nodes that sum over the real and emphemeral connections by replacing expressions like `Add(real_node, ephmeral_node)` with `real_node`.
Then we can clean up the model by taking out orphan nodes and components; I have written the code for this and will commit it at some point in future.